lib/annotations/hooks/AfterBulkCreate.ts

```diff
@@ -0,0 +1,21 @@
+import {IHookOptions} from "../../interfaces/IHookOptions";
+import {addHook} from "../../services/hooks";
+
+export function AfterBulkCreate(target: any, propertyName: string): void;
+export function AfterBulkCreate(options: IHookOptions): Function;
+export function AfterBulkCreate(...args: any[]): void|Function {
+
+  if (args.length === 1) {
```

Comment:
How about this. No changes needed in the hooks service:

```typescript
export function AfterBulkCreate(target: any, propertyName: string): void;
export function AfterBulkCreate(options: IHookOptions): MethodDecorator;
export function AfterBulkCreate(...args: any[]): void|Function {

  return (target: any, propertyName: string) =>
      addHook(target, 'afterBulkCreate', propertyName, args.length ? args[0] : {});
}
```

The polymorphic function signature is confusing, so this keeps the logic related to that close to where the signature is defined instead of in a method somewhere else.